File: ios/ScreenMotion/Models/ThemeConfig.swift
import Foundation

struct ThemeConfig: Equatable, Codable {
    var type: ThemeType = .spaceship
    var gyroEnabled = true
    var touchEnabled = true
    var speed = 1.0
    var particleAmount = 48
    var parallaxStrength = 1.0
    var fps = 30
}

@MainActor
final class ThemeStore: ObservableObject {
    @Published var config = ThemeConfig()
}

File: android/app/src/main/java/com/screenmotion/app/model/ThemeConfig.kt
package com.screenmotion.app.model

data class ThemeConfig(
    val type: ThemeType = ThemeType.SPACE,
    val gyroEnabled: Boolean = true,
    val touchEnabled: Boolean = true,
    val speed: Float = 1f,
    val particleAmount: Int = 48,
    val parallaxStrength: Float = 1f,
    val fps: Int = 30
)
